views/js/qtiCreator/itemCreator.js

```diff
@@ -248,14 +250,26 @@ define([
                             return;
                         }
 
+                        const elementPromises = []; // Prepare to collect promises for asynchronous operations
+
                         _.forEach(item.getComposingElements(), function (element) {
                             if (element.is('customInteraction')) {
                                 usedCustomInteractionIds.push(element.typeIdentifier);
+
+                                if (element.properties && Array.isArray(element.properties.pages) && element.typeIdentifier === 'textReaderInteraction') {
+                                    const pagePromises = xincludeLoader.loadByElementPages(element.properties.pages);
+                                    elementPromises.push(Promise.all(pagePromises).then(updatedPages => {
+                                        element.properties.pages = updatedPages;
+                                    }));
+                                }
```

Comment:
**quibble:** 😱 No, please not add code specific to one interaction that may even not exist... Specific code need to go to specific implementation. If needed, please add the proper adapter pattern.